ArchiSteamFarm.OfficialPlugins.Monitoring/MonitoringPlugin.cs

```diff
@@ -0,0 +1,229 @@
+			try {
+				config = configValue.Deserialize<MonitoringConfig>();
+			} catch (Exception e) {
+				ASF.ArchiLogger.LogGenericException(e);
+				ASF.ArchiLogger.LogGenericWarning(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+				return Task.CompletedTask;
+			}
+		}
+
+		config ??= new MonitoringConfig();
+
+		if (!Enabled) {
+			ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+			return Task.CompletedTask;
+		}
+
+		Config = config;
+
+		ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginInitializedAndEnabled, nameof(MonitoringPlugin)));
+
+		return Task.CompletedTask;
+	}
+
+	public void OnConfiguringEndpoints(IApplicationBuilder app) {
+		ArgumentNullException.ThrowIfNull(app);
+
+		if (!Enabled) {
+			return;
+		}
+
+		app.UseEndpoints(static builder => builder.MapPrometheusScrapingEndpoint());
+	}
+
+	public void OnConfiguringServices(IServiceCollection services) {
+		ArgumentNullException.ThrowIfNull(services);
+
+		if (!Enabled) {
+			return;
+		}
+
+		services.AddOpenTelemetry().WithMetrics(
+			static builder => {
+				builder.AddPrometheusExporter(static config => config.ScrapeEndpointPath = "/Api/metrics");
+				builder.AddMeter("Microsoft.AspNetCore.Hosting");
+				builder.AddMeter("Microsoft.AspNetCore.Server.Kestrel");
+				builder.AddMeter(MeterName);
+			}
+		);
+	}
+
+	public override Task OnLoaded() {
+		Utilities.WarnAboutIncompleteTranslation(Strings.ResourceManager);
+
+		Meter = new Meter(MeterName, Version.ToString());
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_memory_usage",
+			static () => GC.GetTotalMemory(false) / 1024,
+			description: "Current memory usage of ASF"
+		);
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_ipc_banned_ips",
+			static () => ApiAuthenticationMiddleware.GetCurrentlyBannedIPs().Count(),
+			description: "Number of IP addresses currently banned by ASFs IPC module"
+		);
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_active_plugins",
+			static () => PluginsCore.ActivePluginCount,
+			description: "Number of plugins currently loaded in ASF"
+		);
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_bots", static () => {
+				ICollection<Bot> bots = Bot.Bots?.Values ?? ImmutableHashSet<Bot>.Empty;
+
+				return new List<Measurement<int>>(4) {
+					new(bots.Count, new KeyValuePair<string, object?>(TagNames.BotState, "configured")),
+					new(bots.Where(static bot => bot.IsConnectedAndLoggedOn).Count(), new KeyValuePair<string, object?>(TagNames.BotState, "online")),
+					new(bots.Where(static bot => !bot.IsConnectedAndLoggedOn).Count(), new KeyValuePair<string, object?>(TagNames.BotState, "offline")),
+					new(bots.Where(static bot => bot.CardsFarmer.NowFarming).Count(), new KeyValuePair<string, object?>(TagNames.BotState, "farming"))
```

Comment:
Use `Count()` with condition instead of Where + Count, everywhere above too.

Reply:
Funny that Rider did not complain about this.